### PR TITLE
Add Getting Started Notebook

### PR DESCRIPTION
**Pull request by rawkintrevo**
The DCO is a pointless check. It maybe had some reasoning when it was introduced ~15 years ago, but security on github has made it a redundant 'feature'.

/soapbox

This version is a Colab friendly notebook that will help users get started with caikit-nlp. (That has the correct magic words in the commit message :roll_eyes: )

What does Colab have to do with it?
A user can click run in Colab, requisition a T4 GPU for free from google, and run the example.

### CODE REVIEW

**Comment by rawkintrevo** [comment]
For any reviewer that would rather look at the notebook rendering instead of the json rendering convenience link
https://github.com/rawkintrevo/caikit-nlp/blob/101-b/examples/Caikit_Getting_Started.ipynb

File: examples/Caikit_Getting_Started.ipynb [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        "Let's explain each argument in the command:\n",
+        "\n",
+        "1. `caikit-nlp/examples/run_peft_tuning.py`: This specifies the path to the Python script that will be executed. It is a part of the caikit-nlp library and contains the implementation of the PEFT approach.\n",
+        "1. `MULTITASK_PROMPT_TUNING`: This is the option recognized by the `run_peft_tuning.py` script, indicating the type of tuning or learning strategy to be used. In this case, it's \"Multitask Prompt Tuning.\" (The alternative option is `PROMPT_TUNING` which has slightly different options. )\n",
+        "1. `--dataset \"glue/rte\"`: This specifies the dataset to be used for fine-tuning. In this example, the dataset is `glue/rte` which is subset of the GLUE (General Language Understanding Evaluation) benchmark containing the Recognizing Textual Entailment (RTE) task.\n",
+        "1. `--model_name t5-base`: This indicates the base model that will be used for fine-tuning. In this case, it's `t5-base`, which refers to the T5-base model from Hugging Face. **NOTE: At the time of writing the model must exist locally, this is why in the last step we cloned**`t5-base` ** to ** `./t5-base`\n",
```
Comment (alex-jw-brooks) [inline examples/Caikit_Getting_Started.ipynb:125]:
```suggestion
        "1. `--model_name t5-base`: This indicates the base model that will be used for prompt tuning. In this case, it's `t5-base`, which refers to the T5-base model from Hugging Face. **NOTE: At the time of writing the model must exist locally, this is why in the last step we cloned**`t5-base` ** to ** `./t5-base`\n",
```

File: examples/Caikit_Getting_Started.ipynb [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        "1. `--num_epochs 1`: This sets the number of epochs (training iterations) for the fine-tuning process. Here, it's set to 1, meaning the model will go through the dataset once during fine-tuning.\n",
+        "1. ` --verbose`: This is a flag to enable verbose or detailed output during the training process. It will cause the script to print more information about the training progress.\n",
+        "1. `--num_virtual_tokens 100`: This sets the number of virtual tokens used for prompt engineering. Prompt engineering involves adding custom prompts to the data to guide the model's fine-tuning process.\n",
+        "1. `--prompt_tuning_init RANDOM`: This specifies the method used for initializing prompt tuning. In this case, it's set to \"RANDOM,\" meaning the prompt tuning will start with random values.\n",
+        "1. `--output_dir tmp/foo`: This sets the directory where the fine-tuned model and related outputs will be stored. In this case, it's set to the `tmp/foo` directory.\n",
+        "1. `--learning_rate 0.3`: This sets the learning rate for the optimization process during fine-tuning. The learning rate controls the step size in gradient descent.\n",
```
Comment (alex-jw-brooks) [inline examples/Caikit_Getting_Started.ipynb:131]:
```suggestion
        "1. `--learning_rate 0.3`: This sets the learning rate for the optimization process during prompt-tuning. The learning rate controls the step size in gradient descent.\n",
```

**Review by alex-jw-brooks — CHANGES_REQUESTED** [review]
Thanks @rawkintrevo, this looks great! Some small suggestions - thanks for adding this 😄

**Review by alex-jw-brooks — APPROVED** [review]
Awesome - thanks for this, @rawkintrevo! Looks great

**Pull request merged**